server/datastore/mysql/mdm.go

```diff
@@ -163,6 +162,9 @@ FROM (
 		fleetIdentifiers = append(fleetIdentifiers, k)
 	}
 	fleetNamesMap := syncml.FleetReservedProfileNames()
+	for name := range mobileconfig.FleetReservedProfileNames() {
+		fleetNamesMap[name] = struct{}{}
+	}
```

Comment:
Import cycles defeated my attempts to combine the two sets of reserved names so this formulation get repeated in several places throughout this PR.

Reply:
what do you think about this approach? https://github.com/fleetdm/fleet/pull/15394

I'm a bit worried that we'll forget to mix the two maps at some point.